robot/packages/package_path_visitor.go

```diff
@@ -0,0 +1,46 @@
+package packages
+
+import (
+	"path"
+	"reflect"
+
+	"go.viam.com/rdk/config"
+)
+
+// PackagePathVisitor is a visitor that replaces strings containing references to package names
+// with the path containing the package files on the robot.
+type PackagePathVisitor struct {
+	packageManager Manager
+}
+
+// NewPackagePathVisitor creates a new PackagePathVisitor.
+func NewPackagePathVisitor(packageManager Manager) *PackagePathVisitor {
+	return &PackagePathVisitor{
+		packageManager: packageManager,
+	}
+}
+
+// Visit implements config.Visitor.
+func (v *PackagePathVisitor) Visit(data interface{}) (interface{}, error) {
+	t := reflect.TypeOf(data)
+
+	var s string
+	switch {
+	case t.Kind() == reflect.String:
+		s = data.(string)
+	case t.Kind() == reflect.Ptr && t.Elem().Kind() == reflect.String:
+		s = *data.(*string)
+	default:
+		return data, nil
+	}
+
+	ref := config.GetPackageReference(s)
```

Comment:
Just quick note that from first quick glance that should be able to simplify with RefPath helper:

```
// RefPath returns the absolute path of the package reference for a given path with a package reference.
// - If not the original path is not a package reference the original path is returned without an error.
// - If the path contains a package reference and the package does not exist a ErrPackageMissing will be returned.
// - Any syntax errors in the package reference will produce an ErrInvalidPackageRef.

path, err := v.packageManager.RefPath(path)
if err != nil {
	return nil, err
}
```

Reply:
Great suggestion, thank you! Changed this.